### DISCUSSION TITLE
How to render / fetch JSON data

### DISCUSSION THREAD

**Discussion by lpe** [avo-hq/avo#1397] [Help] [answered]
In my Stimulus Controllers I need to fill in a field in a form (create / edit) with informations based on a select. Typically a description based on the product id. This is what I usally do via a change event action :
`...
.fetch(`/products/${id}`)
.then((response) => response.json())
...`
How can I add json rendering (normally in the controller) on Avo without side effects (as I See' You shouldn't need to modify it in order to use Avo.').
Thx by advance for your help.

**Comment by adrianthedev** [comment, marked as answer]
You can add actions to a controller. We should remove that message.

**Reply by lpe** [reply]
Thx. It seems that we ca'nt override the show method with something like :
<pre>
def show
        respond_to do |format|
            format.html ...
            format.json ...
            end
    end
</pre>
Right ? So I suppose I need a specific method to be able to fetch a record in json ?
Thx by advance.

**Reply by adrianthedev** [reply]
Hey @lpe. Apologies for not mentioning it. 

Yes. you would need to create a separate route and controller action.

```ruby
if defined? ::Avo
  Avo::Engine.routes.draw do
    scope :resources do
      get "products/api/index", to: "products#api_index"
    end
  end
end
```

This is just a sample of the controller method. You can go wild or make your own controller.

For v3.0 we're thinking of improving this.

Thank you!
